### PR TITLE
Updated knife bootstrap windows option for consistency

### PR DESCRIPTION
**Pull request by Vasu1105**
1. Updated  --template-file, --distro for deprecation message to be consistent with Chef
2.  Added --bootstrap-template option to be consistent with Chef
3. --bootstrap-template short option -t is conflicting with --winrm-transport short option -t so not defining short option for --bootstrap-template at this time. We need to work on this.

### CODE REVIEW

File: lib/chef/knife/bootstrap_windows_base.rb [hunk 1]
```diff
@@ -127,20 +139,37 @@ def self.included(includer)
         end
       end
 
-      # TODO: This should go away when CHEF-2193 is fixed
+      def default_bootstrap_template
+        "chef-full"
```
Comment (siddheshwar-more) [inline lib/chef/knife/bootstrap_windows_base.rb]: It should be "windows-chef-client-msi" in windows case.

Comment (Vasu1105) [inline lib/chef/knife/bootstrap_windows_base.rb]: Fixed

**Comment by btm** [comment]
Can we avoid the code duplication with core here? Most of load_template looks the same as find_template in core. Besides the methods being organized separated slightly differently, they look the same except for newer encrypted data bag support in core.

```
  def render_template(template=nil)
    if template
      Chef::Log.warn("Passing template to render_template is deprecated in preparation of moving knife-windows into core")
    else
      template = find_template
    end

    if config[:secret_file]
      config[:secret] = Chef::EncryptedDataBagItem.load_secret(config[:secret_file])
    end

    template_content = IO.read(template).chomp
    context = Knife::Core::WindowsBootstrapContext.new(config, config[:run_list], Chef::Config)
    Erubis::Eruby.new(template_content).evaluate(context)
  end   
```

Then inherit to get #bootstrap_template and #find_template, continue to override #default_bootstrap_template?

**Comment by Vasu1105** [comment]
@btm Yes duplication can be avoided. But currently I have done the changes considering current chef version that knife-window master branch is using i.e. Chef 11.

**Comment by btm** [comment]
@Vasu1105 yeah :(. I'm really looking forward to getting knife-windows into core. :+1: for merge.

**Pull request merged**